src/Resolver/TypeResolver/TemplateTypeResolver.php

```diff
@@ -18,7 +18,9 @@ final class TemplateTypeResolver
     public function resolve(Type $type): bool
     {
         if ($type instanceof ObjectType) {
-            return $type->isInstanceOf('Nette\Application\UI\Template')->yes() || $type->isInstanceOf('Nette\Application\UI\ITemplate')->yes();
+            return $type->isInstanceOf('Nette\Application\UI\Template')->yes() ||
+                $type->isInstanceOf('Nette\Application\UI\ITemplate')->yes() ||
+                $type->isInstanceOf('Latte\Runtime\Template')->yes();
```

Comment:
what will change this? We already test rendered templates

Reply:
Use referenced `Nette\Application` tempaltes not `Latte\Runtime` created by `$engine->createTempalte()`

Reply:
I see, I never use it this way

Reply:
Me neither but it can be used this way